### PR TITLE
435 story webpage component connect webpage component to repair sections

### LINKED ISSUE #433: Story: (Webpage Component)  handle logic for flashing selections area
# Template Instructions

Use this template to create a new story item.

1. Copy this template and paste it into a new item project item under backlog.
2. Fill in the placeholders with your own information.
3. Remove these instructions from the item (the horizontal line and above)
4. Add any additional details or context that may be helpful.5. Assign the issue to a team member or yourself.
5. Use the checkboxes to track progress and ensure all requirements are met before marking the issue as complete.
6. Use the checkboxes to track progress and ensure all requirements are met before marking the issue as complete.

---

# <Story Title>

As a **role**, I want **goal**, so that **reason**.

Associated Epic:

## Definition of Done

- [ ] Acceptance criteria defined
- [ ] Solution tasks specified
- [ ] Feature branch created
- [ ] ~~Unit tests written (on API code only for now)~~
- [ ] ~~Solution passes full suite of unit tests~~
- [ ] Design documents updated if needed
- [ ] Pull request created
- [ ] Code is reviewed
- [ ] Code is merged to `develop` branch
- [ ] ~~Tests pass on CI~~
- [ ] ~~Deployed to staging environment~~
- [ ] Acceptance criteria is met

## Acceptance Criteria

- [ ] Given **initial context**, when **event occurs**, then **some desired outcome**
- [ ] Given **initial context**, when **event occurs**, then **some desired outcome**
- [ ] Given **initial context**, when **event occurs**, then **some desired outcome**

## Solution Tasks

- [ ] [Task 1]
- [ ] [Task 2]
- [ ] [Task 3]

### LINKED ISSUE #435: Story: (Webpage Component) Connect Webpage Component to Repair Sections
# Connect Webpage Component to Repair Sections

As a **developer**, I want **to add buttons to the webpage component and connect each button to its respective repair section**, so that **each visual representation connects to the correct repair**.

Associated Epic:

## Definition of Done

- [x] Acceptance criteria defined
- [x] Solution tasks specified
- [x] Feature branch created
- [ ] ~~Unit tests written (on API code only for now)~~
- [ ] ~~Solution passes full suite of unit tests~~
- [x] Design documents updated if needed
- [ ] Pull request created
- [ ] Code is reviewed
- [ ] Code is merged to `develop` branch
- [ ] ~~Tests pass on CI~~
- [ ] ~~Deployed to staging environment~~
- [ ] Acceptance criteria is met

## Acceptance Criteria

- [x] Given **the user wants to fix the navbar**, when **the user clicks the bouncing navbar section**, then **are directed to the navbar repair code block**
- [x] Given **the user wants to fix the date form in the newsletter**, when **the user clicks the bouncing date**, then **they are directed to the dateForm repair code block**
- [x] Given **the user wants to fix the address form**, when **they click the bouncing addressForm section**, then **the user will be directed to the addressForm repair code block**

## Solution Tasks

- [x] Create UI similar to original Figma design for a university newletter
- [x] Create a clickable navigation bar that directs the user to the navbar repair page
- [x] Create a clickable date in newsletter that directs the user to the dateForm repair page
- [x] Create a clickable signup form in newsletter that directs the user to the addressForm repair page
- [x] Make sure each section is clearly visible to the user that it is clickable

### PR DESCRIPTION
**Pull request by dm9965**
- [ ] No discrepancies across browsers. Eg. chrome vs safari
- [ ] Pages pass W3C Validation for HTML (https://validator.w3.org)
- [ ] CSS (https://jigsaw.w3.org/css-validator/)
- [ ] Google sign-in is operational
- [ ] Accessibility functions
- [ ] Pages can scale without distorting page
- [ ] No dead links
- [ ] Navbar is consistent across the site
- [ ] Pages are screen-reader accessible
- [ ] Contrast meets standards for accessibility
- [ ] Pages are keyboard accessible
- [ ] Passes WAVE Evaluation Tool chrome extension

### CODE REVIEW

**Comment by dm9965** [comment]
https://github.com/all-rit/ALL/assets/111752403/ecb4b6f1-9f8a-4283-aaef-b7140635a7cb

File: client/src/components/exercise/lab9/Main.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+ * this will be responsible for iterating through the different stages of the lab
+ * and acting as the container managing the state of the user.
+ */
+const Main = (props) => {
+  const { user } = props;
+  const [exerciseState, setExerciseState] = useState(
+    GAME_STATES.EXERCISE_SELECTION_DEFAULT
+  );
+
+  // useEffect(() => {
```
Comment (AndreasLc1103) [inline client/src/components/exercise/lab9/Main.js]: remove commented out section

**Review by AndreasLc1103 — APPROVED** [review]
Looks good to me 🏁

**Review by AndreasLc1103 — APPROVED** [review]

**Review by AndreasLc1103 — CHANGES_REQUESTED** [review]
Please update branch with changes in `Development/Lab-9`

**Comment by AndreasLc1103** [comment]
this is closed because the code has been merged into other feature branches and will be reviewed then

**Comment by AndreasLc1103** [comment]
This story is going to be expanded to include the did you see it page, we didn't consider this in development and this can be included to quickly wrap up the story. This story will also pull in #461 to get some fixes for database connection and repair management fixes.

**Pull request merged**